enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/helper/CompositeSuspendable.java

```diff
@@ -21,45 +21,35 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 
 import org.neo4j.function.ThrowingConsumer;
 
-public class CompositeEnableable implements Enableable
+public class CompositeSuspendable implements Suspendable
 {
-    private final List<Enableable> enableables = new ArrayList<>();
+    private final List<Suspendable> suspendables = new ArrayList<>();
 
-    public void add( Enableable enableable )
+    public void add( Suspendable suspendable )
     {
-        enableables.add( enableable );
+        suspendables.add( suspendable );
     }
 
     @Override
     public void enable()
     {
-        doOperation( Enableable::enable, "Enable" );
+        doOperation( Suspendable::enable, "Enable" );
     }
 
     @Override
     public void disable()
     {
-        doOperation( Enableable::disable, "Disable" );
+        doOperation( Suspendable::disable, "Disable" );
     }
 
-    private void doOperation( ThrowingConsumer<Enableable,Throwable> operation, String description )
+    private void doOperation( ThrowingConsumer<Suspendable,Throwable> operation, String description )
     {
-        try ( ErrorHandler errorHandler = new ErrorHandler( description ) )
-        {
-            for ( Enableable enableable : enableables )
-            {
-                try
-                {
-                    operation.accept( enableable );
-                }
-                catch ( Throwable throwable )
-                {
-                    errorHandler.add( throwable );
-                }
-            }
-        }
+        ErrorHandler.certainOperations( description, suspendables.stream()
+                .map( (Function<Suspendable,ErrorHandler.ThrowingRunnable>) suspendable -> () -> operation.accept( suspendable ) )
+                .toArray( ErrorHandler.ThrowingRunnable[]::new ) );
```

Comment:
This is quite a complicated way of calling a method. Maybe an overload method on ErrorHandler that takes Collection<T> and Consumer<T>?